### ISSUE TITLE
All widgets should support width & height

### ISSUE THREAD

**Issue by martinohanlon** [lawsie/guizero#136]
At the moment, support it inconsistent.

Should width and height be measured in pixels or text chars? At the moment there is a mixture.

I would say pixels is more intuitive as you already set the app size in pixels.

**Comment by martinohanlon** [comment]
See #132

**Comment by martinohanlon** [comment]
See #59

**Comment by martinohanlon** [comment]
The benefit of setting the size in characters is that it makes the size of the widget dynamic, so if the text size changes the widget automatically resizes.

**Comment by martinohanlon** [comment]
As per #136, a likely implementation for this would be as a shared mixin. There is a SizeMixin in tkmixins which isnt used at the moment and doesnt work.

**Comment by lawsie** [comment]
Yeah, this annoyed me recently when trying to do a grid layout. I think that the sizing in text characters is really weird and not something that would make much logical sense to kids. Even if a widget resizes, the size of the app doesn't so it doesn't seem that helpful unless we can make everything resize?

**Comment by martinohanlon** [comment]
Is it worth making 'height' and 'width' capable of accepting many different forms of input, a bit like the change we made to Color.

e.g.

```
w.width = 10
```

Would use the default for that widget.

```
w.width = '10px' would use pixels
```

This leaves you open to do things like:

```
w.width = 'FILL' 
```

Which would fill it to the size of its container

**Comment by martinohanlon** [comment]
or are percentages more useful e.g. to make it fill a container use `100%`

**Comment by martinohanlon** [comment]
I have committed a change to added width & height to the widgets which dont have it https://github.com/lawsie/guizero/commit/02dfd073a02d9a74e690527d3ce7c36e154fafeb

At the moment I have only implemented the functionality to set width and height in the default way which Tkinter allows. This is a bit inconsistent and some widgets are set in characters and some are set in pixels (in fact the PushButton is set in either characters or pixels depending on whether it has a text label or an image!)

To get all widget to be set in Pixels, we would have to wrap all widgets in a Frame, size the Frame in pixels and stretch the widget to fit the frame.  This is clearly a bigger change, so i will create a new issue about "allowing widgets size to be set in pixels or percentages".
